### PR TITLE
gator 3.9.0 (new formula)

### PR DESCRIPTION
**Pull request by salaxander**
gator is the cli tool for Open Policy Agent Gatekeeper. This PR adds the cli to be available via brew.

Apologies for the previous incomplete PR. This is now a source package rather than a binary as requested, and I've properly run all audit commands locally prior to opening.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/gator.rb
Comment (carlocab) [inline Formula/gator.rb:6]: Align the arguments to `url`:
```suggestion
  url "https://github.com/open-policy-agent/gatekeeper.git",
      tag:      "v3.9.0",
      revision: "dfd2dd80734e20d30c00f7e729f03ef5ee9730ed"
```
Do we need a Git clone? Can we use a tarball?

File: Formula/gator.rb
Comment (carlocab) [inline Formula/gator.rb:18]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/gator.rb
Comment (carlocab) [inline Formula/gator.rb:14]: Can we use
```ruby
system "go", "build", *std_go_args
```
?

File: Formula/gator.rb
Comment (salaxander) [reply Formula/gator.rb:6]: There is a source tarball - I ended up going this route because looking at other go based formulas, this seemed to be the pattern (kubectl, kubectx, k9s). I can certainly switch it though if that's preferred.

File: Formula/gator.rb
Comment (salaxander) [reply Formula/gator.rb:14]: Yeah I can switch to that. Went this way for the built in ldflags with the make target, but I can also do it with go build

**Comment by salaxander** [comment]
@carlocab Thanks for the comments! Converted this from git based to tarball, switched to system go build, and added a more comprehensive test

File: Formula/gator.rb
Comment (SMillerDev) [inline Formula/gator.rb:93]: Otherwise this would match anything
```suggestion
    assert_equal "", shell_output("#{bin}/gator test -f gator-manifest.yaml -f template-and-constraints/")
```

Comment (carlocab) [reply Formula/gator.rb:93]: Better:
```suggestion
    assert_empty shell_output("#{bin}/gator test -f gator-manifest.yaml -f template-and-constraints/")
```

File: Formula/gator.rb
Comment (carlocab) [inline Formula/gator.rb:14]: Let's check the output of `--version` in the `test` block to make sure that this `-X` flag is valid.

Comment (salaxander) [reply Formula/gator.rb:14]: currently that won't actually work - the tool itself is still outputting alpha as the version, but the intention is to utilize the ldflag in the future, so it's still passed in. I actually asked the maintainer about this as well and that's the answer for the moment

**Comment by salaxander** [comment]
@carlocab @SMillerDev Would one of you be able to take another look at this?

Thank you!

**Comment by salaxander** [comment]
@carlocab @SMillerDev Sorry for the additional ping - could I get another review on this?

**Review by carlocab — APPROVED** [review]
FYI, typical GitHub etiquette (or, at least, for Homebrew repositories) is to wait at least a week between pings.

Removing review requests in order to re-request them is also, in my view, abuse of the feature.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@salaxander, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**
